framework/i18n/MessageFormatter.php

```diff
@@ -274,19 +275,20 @@ protected function fallbackFormat($pattern, $args, $locale)
      */
     private static function tokenizePattern($pattern)
     {
+        $charset = Yii::$app ? Yii::$app->charset : 'UTF-8';
```

Comment:
I am unsure about this: i18n feature requires usage of 'UTF-8' encoding no matter of application encoding.